Wikipedia/Code/UIViewController+WMFArticlePresentation.m

```diff
@@ -42,10 +42,12 @@ - (void)wmf_pushArticleViewController:(WMFArticleViewController *)viewController
     }
     [[PiwikTracker wmf_configuredInstance] wmf_logView:viewController];
 
-    dispatchOnMainQueueAfterDelayInSeconds(0.5, ^{
-        MWKHistoryList *historyList = viewController.dataStore.userDataStore.historyList;
-        [historyList addPageToHistoryWithURL:viewController.articleURL];
-    });
+    if(viewController.isAddingArticleToHistoryListEnabled){
+        dispatchOnMainQueueAfterDelayInSeconds(0.5, ^{
```

Comment:
It was there to prevent the history view controller from updating before pushing the article VC. It can b removed if there if:
1. Open the history tab
2. Tap on an older item
3. The article pushes in without the history tableview re-ordering during the push animation 

cc @joewalsh @montehurd